UnitTests/Common/toolboxes/ndflattener_tests.cpp

```diff
@@ -51,6 +56,8 @@ TEST_CASE("NdFlattener Test", "[NdFlattener]"){
   NdFlattener<2> nd2;
   nd2.initialize_or_refresh(f);
 
+  setIntoConstObject(nd2[0]);
```

Comment:
Do you mean this:

    const NdFlattener<2>& nd2_const = nd2;
    setIntoConstObject(nd2_const[0]);

Reply:
No, I mean exactly what I wrote.
If you have any object and you pass it to a function by const& that function cannot be able to change it.
You need const and non-const versions of operator [] in the Accumulator class.

Reply:
The statement `obj[0] = 1;` does not change the const `IndexAccumulator obj`, it changes the non-const `NdFlattener obj.nd`. So IMO it is perfectly fine that the code compiles. If `nd2` was const, the code would not compile. This is precisely the behaviour that I intended.

Are you saying that this behaviour is not good? I.e., that a function with a const argument `obj` that implements the [] interface should never be allowed to do `obj[0]=1;`, no matter what `operator[]` actually does? 

If yes, I'm fine with changing this. As you say, essentially there should be both methods `operator[]`, `operator[] const` returning `LookupType&` and `LookupType const&`, respectively (and the same for `data`). But let's discuss this in the developer meeting tomorrow.

Reply:
To be clear, I think that behavior is unsafe and incorrect.
Constness should be derived from current context and not from how the object was created initially.
If a function is allowed to modify the state of something that is supposed to be read only, it is not const-correct and it is very difficult to guarantee thread safety.
Please change it so we can move on, enough time has been put into this already.

Reply:
Ok. f0e16b0 failed to build, as expected. I removed the forbidden write access in 722a4dc.

You're mentioning thread safety, should we put `SU2_OMP_MASTER { ... } END_SU2_OMP_MASTER` around the code in `initialize_or_refresh`?

Reply:
Thanks. No we leave that for the users of the class otherwise thread local objects would not be possible.